### DISCUSSION TITLE
[Feature]: Allow configuration of credentials in useFetcher submit

### DISCUSSION THREAD

**Discussion by alexandesigner** [remix-run/react-router#12023] [Proposals]
### What problem does this feature solve?

Currently, the ``useFetcher`` hook from React Router provides a convenient way to handle form submissions in a declarative manner. However, there is no direct way to configure options such as credentials ``(e.g., 'include', 'same-origin', 'omit')`` when calling the submit function.

This limitation makes it difficult to handle scenarios where credentials (such as cookies or authentication tokens) need to be included with the request, especially when interacting with APIs that require ``credentials: 'include'``.

### What does the proposed API look like?

Add support for passing fetch options (specifically ``credentials``) in the submit function of the ``useFetcher`` hook. This would allow developers to configure requests with greater flexibility when needed.

```ts
const fetcher = useFetcher();

// Current behavior (no way to set credentials):
fetcher.submit(formData, { method: 'post', action: '/submit' });

// Proposed behavior (ability to set credentials):
fetcher.submit(formData, { method: 'post', action: '/submit', credentials: 'include' });
```

**Comment by kiliman** [comment]
Fetchers are designed to call your RR routes. They are not meant to replace `fetch` calls to external APIs.
